### PR TITLE
[plugin-db] Add step to merge two datasets

### LINKED ISSUE #463: Provide possibility to merge data from different tables
In order to test data marts properly, there should be a possibility to merge results of execution of different queries against different databases into one table

### PR DESCRIPTION
**Pull request by ikalinin1**

### CODE REVIEW

**Comment by ikalinin1** [comment]
I'll add system test for these changes, ones I understand how to up Postgres on Heroku

File: vividus-plugin-db/src/main/java/org/vividus/bdd/steps/db/DatabaseSteps.java [hunk 1]
```diff
@@ -136,6 +137,37 @@ public void executeSql(String sqlQuery, String dbKey, Set<VariableScope> scopes,
...
+    public void joinDataSets(List<Map<String, Object>> left, List<Map<String, Object>> right,
+            Set<VariableScope> scopes, String variableName)
+    {
+        List<Map<String, Object>> result = new ArrayList<>(left.size() + right.size());
+        if (!left.isEmpty() && !right.isEmpty())
+        {
+            Set<String> leftHeader = left.get(0).keySet();
+            Set<String> rightHeader = right.get(0).keySet();
+            Validate.isTrue(leftHeader.equals(rightHeader),
+                    "Data sets should have same columns;\nLeft:   %s\nRight: %s", leftHeader, rightHeader);
```
Comment (valfirst) [inline vividus-plugin-db/src/main/java/org/vividus/bdd/steps/db/DatabaseSteps.java:164]: `Left:   %s` - tripled space

Comment (ikalinin1) [reply vividus-plugin-db/src/main/java/org/vividus/bdd/steps/db/DatabaseSteps.java:164]:
```
Left:  [1, 2, 3]
Right: [1, 2, 3]
```

Comment (ikalinin1) [reply vividus-plugin-db/src/main/java/org/vividus/bdd/steps/db/DatabaseSteps.java:164]: Two is enough you're right

File: vividus-plugin-db/src/main/java/org/vividus/bdd/steps/db/DatabaseSteps.java [hunk 2]
```diff
@@ -136,6 +137,37 @@ public void executeSql(String sqlQuery, String dbKey, Set<VariableScope> scopes,
...
+    public void joinDataSets(List<Map<String, Object>> left, List<Map<String, Object>> right,
+            Set<VariableScope> scopes, String variableName)
+    {
+        List<Map<String, Object>> result = new ArrayList<>(left.size() + right.size());
+        if (!left.isEmpty() && !right.isEmpty())
+        {
+            Set<String> leftHeader = left.get(0).keySet();
+            Set<String> rightHeader = right.get(0).keySet();
+            Validate.isTrue(leftHeader.equals(rightHeader),
+                    "Data sets should have same columns;\nLeft:  %s\nRight: %s", leftHeader, rightHeader);
```
Comment (uarlouski) [inline vividus-plugin-db/src/main/java/org/vividus/bdd/steps/db/DatabaseSteps.java:164]: please mention this in docs

**Pull request merged**